### ISSUE TITLE
Migrate `eslint-config-next` to `eslint-plugin-import-x` and `eslint-react`

### ISSUE THREAD

**Issue by mcmxcdev** [43081j/ecosystem-cleanup#123]
I just saw that `eslint-config-next` pulls in `eslint-plugin-import`, `eslint-plugin-jsx-a11y` and `eslint-plugin-react`. These dependencies are known to be full of bloat dependencies, polyfills and support for ancient node versions.

See: https://github.com/vercel/next.js/blob/b68c188450f1d4ed932e9e28e9be2b5562bfe306/packages/eslint-config-next/package.json#L19-L21

Although probably not a 100% replacement in terms of rulesets, I would suggest moving from:
- `eslint-plugin-import` -> https://github.com/un-ts/eslint-plugin-import-x
- `eslint-plugin-react` -> https://github.com/Rel1cx/eslint-react
- `eslint-plugin-jsx-a11y` doesn't have an equivalent replacement afaik, but https://github.com/Rel1cx/eslint-react/issues/335 is talking about a future `eslint-plugin-react-dom-a11y` plugin

**Comment by 43081j** [comment]
Good find 👍

Id love to see a jsx-a11y alternative too. As far as I know, nobody has built one yet

**Comment by mcmxcdev** [comment]
Biome does have accessibility rules ported over from `eslint-plugin-jsx-a11y`, see https://biomejs.dev/linter/rules/#accessibility but projects would need to be open to embrace another tool rather than just another eslint plugin.

**Comment by 43081j** [comment]
Yup I think we still need to provide a solution for eslint users

We can recommend both but they will be a preference for many years to come. So we need to make sure we satisfy both sides
